src/kubernetes_cluster/spec/kubernetes_api/state_machine.rs

```diff
@@ -176,6 +180,8 @@ pub open spec fn validate_update_request(req: UpdateRequest, s: KubernetesAPISta
         && req.obj.metadata.resource_version != s.resources[req.key].metadata.resource_version {
         // Update fails because the object has a wrong rv
         Option::Some(APIError::Conflict)
+    } else if req.obj.kind == Kind::CustomResourceKind && !K::transition_rule(req.obj, s.resources.index(req.key)) {
+        Option::Some(APIError::IllegalCustomResource)
```

Comment:
Is it the same error that K8s will return in real execution if validation cannot pass?